ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -194,28 +232,28 @@ FileSpan addPutEntries(int count, long size, long expiresAtMs) throws IOExceptio
       } else {
         liveKeys.add(id);
       }
+      index.addToIndex(entry, fileSpan);
+      lastModifiedTimesInSecs.put(indexSegmentStartOffset, time.seconds());
       expectedJournalLastOffset = fileSpan.getStartOffset();
       endOffsetOfPrevMsg = fileSpan.getEndOffset();
     }
-    FileSpan fileSpan = new FileSpan(indexEntries.get(0).getValue().getOffset(), endOffsetOfPrevMsg);
-    index.addToIndex(indexEntries, fileSpan);
     assertEquals("End Offset of index not as expected", endOffsetOfPrevMsg, index.getCurrentEndOffset());
     assertEquals("Journal's last offset not as expected", expectedJournalLastOffset, index.journal.getLastOffset());
-    return fileSpan;
+    return indexEntries;
   }
 
   /**
    * Adds a delete entry in the index (real and reference) for {@code idToDelete}.
    * @param idToDelete the id to be deleted.
    * @return the {@link FileSpan} of the added entries.
+   * @throws InterruptedException
    * @throws IOException
    * @throws StoreException
    */
-  FileSpan addDeleteEntry(MockId idToDelete) throws IOException, StoreException {
+  FileSpan addDeleteEntry(MockId idToDelete) throws InterruptedException, IOException, StoreException {
```

Comment:
Could you elaborate? `addPutEntries()` returns a list because it can add more than one entry. This function can add only entry at a time and so does not make sense to return a list.